templates/trust/contracts/TrustTemplate.sol

```diff
@@ -114,19 +95,30 @@ contract TrustKit is KitBase, IsContract {
 
         Kernel dao = _getDaoCache(msg.sender);
         MultiSigWallet multiSig = _setupMultiSig(dao, multiSigKeys);
-        emit DeployTrustEntity(address(dao), address(multiSig));
+        emit DeployMultiSig(address(multiSig));
         return multiSig;
     }
 
     function _setupApps(Kernel dao, address[] beneficiaryKeys, address[] heirs, uint256[] heirsStake, uint256 blockedHeirsSupply) internal {
         ACL acl = ACL(dao.acl());
-        acl.createPermission(address(this), dao, dao.APP_MANAGER_ROLE(), address(this));
+        Agent agent = _installAgentApp(dao);
+        Vault vault = _installVaultApp(dao);
+        Finance finance = _installFinanceApp(dao, vault, 30 days);
+        (Voting holdVoting, Voting heirsVoting, TokenManager holdTokenManager, TokenManager heirsTokenManager) = _installTokenApps(dao);
+
+        _mintHoldTokens(acl, holdTokenManager, beneficiaryKeys);
+        _mintHeirsTokens(acl, heirsTokenManager, heirs, heirsStake, blockedHeirsSupply);
+
+        _createVaultPermissions(acl, vault, finance, holdVoting);
+        _createFinancePermissions(acl, finance, holdVoting, holdVoting);
+        _createTokenManagerPermissions(acl, holdTokenManager, holdVoting);
+        _createTokenManagerPermissions(acl, heirsTokenManager, heirsVoting);
+        _createEvmScriptsRegistryPermissions(acl, holdVoting, holdVoting);
+        _createVotingPermissions(acl, holdTokenManager, holdVoting);
+        _createVotingPermissions(acl, heirsTokenManager, heirsVoting);
+        _createAgentPermission(acl, agent, agent.EXECUTE_ROLE(), holdVoting, heirsVoting);
+        _createAgentPermission(acl, agent, agent.RUN_SCRIPT_ROLE(), holdVoting, heirsVoting);
```

Comment:
Could we group these Voting and Agent permissions into their own custom functions in this template?

Reply:
yup, I was waiting to merge some other PRs cause I was planning to leverage other helpers 😏